homeassistant/components/media_player/samsungtv.py

```diff
@@ -0,0 +1,168 @@
+"""
+homeassistant.components.media_player.denon
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Provides an interface to Samsung TV with a Laninterface.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.samsungtv/
+"""
+import logging
+import socket
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice, SUPPORT_PAUSE, SUPPORT_VOLUME_STEP,
+    SUPPORT_VOLUME_MUTE, SUPPORT_PREVIOUS_TRACK,
+    SUPPORT_NEXT_TRACK, SUPPORT_TURN_OFF,
+    DOMAIN)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME, STATE_OFF,
+    STATE_ON, STATE_UNKNOWN)
+
+CONF_PORT = "port"
+CONF_TIMEOUT = "timeout"
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['samsungctl==0.5.1']
+
+SUPPORT_SAMSUNGTV = SUPPORT_PAUSE | SUPPORT_VOLUME_STEP | \
+    SUPPORT_VOLUME_MUTE | SUPPORT_PREVIOUS_TRACK | \
+    SUPPORT_NEXT_TRACK | SUPPORT_TURN_OFF
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the Samsung TV platform. """
+
+    if not config.get(CONF_HOST):
+        _LOGGER.error(
+            "Missing required configuration items in %s: %s",
+            DOMAIN,
+            CONF_HOST)
+        return False
+    name = config.get("name", 'Samsung TV Remote')
+    # Generate a config for the Samsung lib
+    remote_config = {
+        "name": "HomeAssistant",
+        "description": config.get(CONF_NAME, ''),
+        "id": "ha.component.samsung",
+        "port": config.get(CONF_PORT, 55000),
+        "host": config.get(CONF_HOST),
+        "timeout": config.get(CONF_TIMEOUT, 0),
+    }
+
+    add_devices([SamsungTVDevice(name, remote_config)])
+
+
+# pylint: disable=abstract-method
+class SamsungTVDevice(MediaPlayerDevice):
+    """ Represents a Samsung TV. """
+
+    # pylint: disable=too-many-public-methods
+    def __init__(self, name, config):
+
+        self._name = name
+        # Assume that the TV is not muted
+        self._muted = False
+        # Assume that the TV is in Play mode
+        self._playing = True
+        self._state = STATE_UNKNOWN
+        self._remote = None
+        self._config = config
+
+    def update(self):
+        # Send an empty key to see if we are still connected
+        return self.send_key('KEY_POWER')
+
+    def get_remote(self):
+        """ Creates or Returns a remote control instance """
+        from samsungctl import Remote
```

Comment:
I would recommend doing this once in the `__init__` method and storing it at the class level. That prevents you from having to do it more than once. With this implementation, the import is run twice each time `send_key` is called.